feat: add adaptive text button

PouriaMoradi021 opened the pull request.

PouriaMoradi021 committed:
refactor: make every title in adaptive primary button multilingual support
- update `AdaptiveButtonContent`

PouriaMoradi021 committed:
refactor: re-arrange imports files to create access from out of this package
- update & rename `app_core`
- update & rename `app_widgets`

PouriaMoradi021 committed:
refactor: refactor `AdaptivePrimaryButton` widget
- update `getFluentButtonStyleMethod`
- add `isDisabled` & remove `isOutlined` from adaptive primary button use case
- update `AdaptivePrimaryButton` tests
- update `AdaptivePrimaryButton` & add MouseRegion to it

PouriaMoradi021 committed:
style: reformat codes with dart reformat command

PouriaMoradi021 committed:
feat: impl `Adaptive Secondary Button` widget & it`s use cases
- add/create `AdaptiveSecondaryButton`
- add/create `adaptiveSecondaryButtonUseCase` method
- add/create `getFluentOutlinedButtonStyleMethod`
- update `AppColors` & add `primaryBlackColor` to it

PouriaMoradi021 committed:
test: add tests for `AdaptiveSecondaryButton` in different states

PouriaMoradi021 committed:
docs: add documents according to mark down standards

PouriaMoradi021 committed:
style: re-arrange imports & reformat codes with dart reformat command

PouriaMoradi021 committed:
feat: impl `AdaptiveTextButton` widget & it`s use case
- add/create `AdaptiveTextButton`
- add/create `adaptiveTextButtonUseCase`
- update mai.directories file

PouriaMoradi021 committed:
test: add tests for `AdaptiveTextButton` in different states

PouriaMoradi021 committed:
docs: add documents according to mark down standards

PouriaMoradi021 committed:
style: re-arrange imports & reformat codes with dart reformat command

PouriaMoradi021 committed:
conflicts: resolve conflicts with `main` branch
- approved by @PouriaMoradi021

PouriaMoradi021 committed:
docs: update `Changelog.md` & `pubspec.yaml` files

PouriaMoradi021 committed:
docs: update `Changeloge.md` & `pubspec.yaml` files

esmaeil-ahmadipour commented:
@PouriaMoradi021  , Text button colors need flexibility in dark/light theme. please test it and resolve this issue.

PouriaMoradi021 committed:
fix: correct foreground color in adaptive text button
- remove conditions in foreground color & remove `isDisables` knob from it`s use case

PouriaMoradi021 committed:
Merge branch 'feat/add-adaptive-text-button' of https://github.com/pactus-project/pactus-gui-widgetbook into feat/add-adaptive-text-button

PouriaMoradi021 committed:
docs: update `changelog.md` file

PouriaMoradi021 committed:
fix: add `isDefaultTextButton` to `AdaptiveTextButton`
- update AdaptiveTextButton class & it`s use case

PouriaMoradi021 committed:
docs: update `changelog.md` file

PouriaMoradi021 commented:
> @PouriaMoradi021 , Text button colors need flexibility in dark/light theme. please test it and resolve this issue.

it`s Done!
